packages/site/site-react/src/iframebridge/useBlockPreviewFetch.tsx

```diff
@@ -0,0 +1,41 @@
+import { useEffect, useState } from "react";
+
+import { createFetchInMemoryCache } from "../graphQLFetch/fetchInMemoryCache";
+import { type GraphQLFetch, convertPreviewDataToHeaders, createFetchWithDefaults, createGraphQLFetch } from "../graphQLFetch/graphQLFetch";
+import { useIFrameBridge } from "./useIFrameBridge";
+
+const cachingFetch = createFetchInMemoryCache(fetch);
+
+type Fetch = typeof fetch;
+
+/**
+ * @deprecated Use useBlockPreviewFetch() without argument instead
+ * @param apiUrl the API URL
+ */
+export function useBlockPreviewFetch(apiUrl: string): { fetch: Fetch; graphQLFetch: GraphQLFetch };
+export function useBlockPreviewFetch(apiUrl?: string | undefined): { fetch: Fetch; graphQLFetch?: GraphQLFetch };
+export function useBlockPreviewFetch(apiUrl?: string | undefined) {
+    const { showOnlyVisible, graphQLApiUrl } = useIFrameBridge();
+    const [graphQLFetch, setGraphQLFetch] = useState<GraphQLFetch | undefined>(() =>
+        apiUrl ? createBlockPreviewFetch(apiUrl, !showOnlyVisible) : undefined,
+    );
+
+    useEffect(() => {
+        if (graphQLApiUrl) {
+            // We need to use an updater function here because createBlockPreviewFetch's return value would otherwise be incorrectly treated as an updater function.
+            setGraphQLFetch(() => createBlockPreviewFetch(graphQLApiUrl, !showOnlyVisible));
+        }
+    }, [showOnlyVisible, graphQLApiUrl]);
+
+    return {
+        graphQLFetch,
+        fetch: cachingFetch,
+    };
+}
+
+function createBlockPreviewFetch(graphqlApiUrl: string, includeInvisible: boolean) {
+    return createGraphQLFetch(
+        createFetchWithDefaults(cachingFetch, { headers: convertPreviewDataToHeaders({ includeInvisible }), credentials: "include" }),
```

Comment:
useBlockPreviewFetch now exists in site-react and site-nextjs because it uses `createFetchWithDefaults` which is framework-specific

Reply:
I think you don't have to duplicate this file.

While it is correct that there is a framework-specific `createFetchWithDefaults`, you don't need to use the nextjs-specific implementation, because you don't pass `next` to `createFetchWithDefaults`.

Reply:
It isn't exported in src/index.ts, was this intentional?

Reply:
I also removed createFetchWithPreviewHeaders since here also next isn't passed to createFetchWithDefaults: 55b136f6af8b31ca8d706e3594ed4892d499e607